crates/runtime/src/instance/allocator/pooling/unix.rs

```diff
@@ -0,0 +1,26 @@
+use crate::Mmap;
+
+pub unsafe fn make_accessible(addr: *mut u8, len: usize) -> bool {
+    region::protect(addr, len, region::Protection::READ_WRITE).is_ok()
+}
+
+pub unsafe fn decommit(addr: *mut u8, len: usize) {
```

Comment:
I'm fine with doing that.

With `uffd` enabled, perhaps we just won't call these functions rather than stubbing them out (`decommit` for uffd is very similar, except it doesn't modify any protection levels).

Reply:
Eh either way is fine by me, I'm mostly just looking to consolidate platform-specific memory manipulation. The `Mmap` type already is a bit odd I think in that it's kinda using an upstream crate kinda not, and it may be best if we just make our own module nowadays. Anyway that can be a refactoring for another day.

Reply:
So I've refactored this a little in a forthcoming push that makes `make_accessible` less generic sounding and explicitly related to protecting pages for linear memories (this also makes the semantics clearer when uffd is used).  `decommit` remains, but I don't think belongs on `Mmap` as the pages to decommit are rarely the entire mapping (the pooling alloctor tries to release the backing of a pages that were "used" for the instance, ensuring that new zero pages are mapped back in for the next instance).

As such, I don't think there's much to do for this comment and I'm resolving it.